File: Difficulty: Easy/Sum of all prime numbers between 1 and N./sum-of-all-prime-numbers-between-1-and-n..cpp
//{ Driver Code Starts
//Initial Template for C++

#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends

//User function Template for C++

class Solution{
	public:
   	long long int prime_Sum(int n){
        vector<bool> sieve(n+1,true);
        sieve[0]=sieve[1] = false;
        long long int sum =0;
        for(int i=2;i*i<=n;i++)
        {
            if(sieve[i])
            {
                
                for(int j=i*i;j<=n;j+=i)
                {
                    sieve[j]= false;
                }
            }
        }
      for(int i=2;i<=n;i++)
       {
           if(sieve[i])
           {
               sum+=i;
           }
       }
       return sum;
   	}    
};


//{ Driver Code Starts.
int main(){
	int tc;
	cin >> tc;
	while(tc--){
		int n;
		cin >> n;
		Solution ob;
		long long int ans = ob.prime_Sum(n);
		cout << ans <<"\n";
	
cout << "~" << "\n";
}  
	return 0;
}
// } Driver Code Ends

File: Difficulty: Easy/Sum of all prime numbers between 1 and N./sum-of-all-prime-numbers-between-1-and-n..java
//{ Driver Code Starts
import java.util.*;
import java.lang.*;
import java.io.*;
class GFG
{
    public static void main(String[] args) throws IOException
    {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine().trim());
        while(T-->0)
        {
            int n = Integer.parseInt(br.readLine().trim());
            Solution ob = new Solution();
            long ans = ob.prime_Sum(n);
            System.out.println(ans);
        
System.out.println("~");
}
    }
}

// } Driver Code Ends


class Solution
{
    public long prime_Sum(int n)
    {
        long result =0;
        boolean[] sieve = new boolean[n+1];
        Arrays.fill(sieve,true);
        sieve[0]=sieve[1] = false;
        for(int i =2;i*i<=n;i++)
        {
            if(sieve[i])
            {
                
                for(int j= i*i;j<=n;j+=i)
                {
                    sieve[j] = false;
                }
            }
        }
        
        for(int i=0;i<=n;i++)
          {
              if(sieve[i])
                 result+=i;
          }
        return result;
    }
}